### PR TITLE
[ESNext] Add some new proposals

### PR DESCRIPTION
**Pull request by zloirock**
- [`Object.fromEntries`](https://github.com/tc39/proposal-object-from-entries)
- [Getting last item from `Array`](https://github.com/keithamus/proposal-array-last)
- [New `Set` methods](https://github.com/tc39/proposal-set-methods)
- [New collections (`Set` and `Map`) methods](https://github.com/tc39/proposal-collection-methods)

### CODE REVIEW

File: data-esnext.js [hunk 1]
```diff
@@ -3483,6 +3483,261 @@ exports.tests = [
...
+    {
+      name: 'Array.prototype.lastItem',
+      exec: function () {/*
+        return [1, 2, 3].lastItem === 3;
+      */},
+      res: {
+      }
+    },
+    {
+      name: 'Array.prototype.lastIndex',
```
Comment (ljharb) [inline data-esnext.js:3513]: this is highly likely to go away, fwiw

**Review by ljharb — APPROVED** [review]

File: data-esnext.js (hunk 1 above)
Comment (zloirock) [reply data-esnext.js:3513]: I don't see anything about it in the repo of the proposal.

Comment (zloirock) [reply data-esnext.js:3513]: But I agree, I don't see any use cases for this getter.

Comment (ljharb) [reply data-esnext.js:3513]: you're right, there's nothing in there yet. but `[].lastIndex` being `0` is super weird, and some on the committee were uncomfortable with it.

**Pull request merged**